solr/core/src/java/org/apache/solr/update/processor/TimeRoutedAliasUpdateProcessor.java

```diff
@@ -262,8 +304,14 @@ private boolean updateParsedCollectionAliases() {
     return false;
   }
 
-  /** Given the route key, finds the collection.  Returns null if too old to go in last one. */
-  private String findTargetCollectionGivenTimestamp(Instant docTimestamp) {
+  /**
+   * Given the route key, finds the correct collection or returns the most recent collection if the doc
+   * is in the future. Future docs will potentially cause creation of a collection that does not yet exist
+   * or an error if they exceed the maxFutureMs setting.
+   *
+   * @throws SolrException if the doc is too old to be stored in the TRA
+   */
+  private String findCandidateCollectionGivenTimestamp(Instant docTimestamp, String id) {
```

Comment:
Why the "candidate" terminology?  That word, to me, suggests something that is tentative -- that the ultimate choice may be something different.